### DISCUSSION TITLE
SH 1106 SPI Reset Pin = +3.3V?

### DISCUSSION THREAD

**Discussion by Fishbone69** [olikraus/u8g2#2124] [General]
Hi all!  I know Oli reads all of these posts so I am hoping he can add some nuance to what I am saying.  If you read this:https://github.com/olikraus/u8g2/discussions/2116 you will see I had an issue getting the SH1106 SPI oled to work because I thought the reset pin was "optional".  It is not.  However I went back to a data sheet for the oled and it says the following:
"Power Reset for Controller and Driver This pin is reset signal input. When the pin is low, initialization of the chip is executed. Keep this pin pull high during normal operation."  I connected mine to my +3.3V and it seems to be operating normally.  I even put it through some sleep/wake cycles and it seems to work fine.  I do not claim to know whether the initialization is being handled by the software - which is where Oli can probably shed some light.  The reason I am sharing this is because SPI uses a lot of gpio pins and if I can save anybody a pin with this then it's worth it.  I'll update this post if I notice any anomalous behavior.  Also, I have read a post where this didn't work on a SSD1306 so mileage may vary. 

Fish

**Comment by olikraus** [comment]
The datasheet also tells you to hold the reset line low for 10us (page 44). So if you just connect the reset line to +3.3V, then the reset line goes high together with the power supplier. However (of course) the 10us low peak is meant to be there while power is up. So the correct timing during startup would be: Power goes up to +3.3V, then, with a delay of 10us, voltage level for reset goes up to +3.3V.

U8g2 will just do this delay and then the reset line will state at +3.3V forever (unless you reexecute .begin()).

In order to reduce GPIO count you could connect your reset line to +3.3 volt and it may work, but actually you are violating the datasheet specification. The proper option to avoid the GPIO pin, is to create an analog RC delay for the reset input of the display.

For some details see here: https://electronics.stackexchange.com/questions/387523/delayed-active-low-short-pulse-for-reset

In the above link, the delay will be 10ms, which is 1000 times more than required (SH1106 requires a 10us pulse), so you can reduce the capacitor value a lot. 0.1uF should be ok instead.

**Reply by Fishbone69** [reply]
yeah, that's no issue.  Although I was going to take a crack at doing my own pcb.  Page 8 of the attached has resistors R1, R3, R6, R7, and R8.  To switch from SPI to I2C you put R1 on R3.  But does this also imply R8, R7, and R8 are not required if you are only running SPI? (actually, there is no R8 in the photo.)
[Datasheet SH1106.pdf](https://github.com/olikraus/u8g2/files/10913910/Datasheet.SH1106.pdf)

**Comment by olikraus** [comment]
I think you only have to move that one resistor
